include/uart.h

```diff
@@ -257,6 +318,60 @@ static inline unsigned char _impl_uart_poll_out(struct device *dev,
 	return api->poll_out(dev, out_char);
 }
 
+/**
+ * @brief Set UART configuration.
+ *
+ * Sets UART configuration using data from *cfg.
+ *
+ * @param dev UART device structure.
+ * @param cfg UART configuration structure.
+ *
+ *
+ * @retval -ENOTSUP If configuration is not supported by device.
+ *                  or driver does not support setting configuration in runtime.
+ * @retval 0 If successful, negative errno code otherwise.
+ */
+__syscall int uart_configure(struct device *dev, const struct uart_config *cfg);
+
+static inline int _impl_uart_configure(struct device *dev,
+				       const struct uart_config *cfg)
+{
+	const struct uart_driver_api *api =
+				(const struct uart_driver_api *)dev->driver_api;
+
+	if (api->configure) {
+		return api->configure(dev, cfg);
+	}
+
+	return -ENOTSUP;
+}
+
+/**
+ * @brief Get UART configuration.
+ *
+ * Stores current UART configuration to *cfg, can be used to retrieve initial
+ * configuration after device was initialized using data from DTS.
+ *
+ * @param dev UART device structure.
+ * @param cfg UART configuration structure.
+ *
+ * @retval -ENOTSUP If driver does not support getting current configuration.
+ * @retval 0 If successful, negative errno code otherwise.
+ */
+__syscall int uart_get_config(struct device *dev, struct uart_config *cfg);
```

Comment:
nit, we have a mix of API function naming conventions, sometime verb at the end, sometime in the middle, i have no preference, but uart_config_get reads better. We should try and standarize this for new APIs

Reply:
> but uart_config_get reads better.

Well, that's hardly. And within 2 functions added here, there's a consistency: first verb goes, then optional object (noun). (`configure` is a verb). It's inconsistent with the older API, yeah.

> We should try and standarize this for new APIs

We should try to standardize, or should stick for consistency to whatever semi-random and not standardized way it was done before?

Reply:
Changed to uart_config_get

Reply:
@pfalcon 
> Well, that's hardly.

Obviously that is very subjective. Maybe because I read from right to left.

This is how we have thing in the kernel API and in some drivers.